server/daos/customers.js

```diff
@@ -0,0 +1,131 @@
+export const checkCabAvailability = async (cabId)=>{
+    //Check already reserved this requested cab
+    let cabReserved = await db.bookings.findOne({
+        attributes:['vehicleId','id','amount'],
+        where:{
+            status : 'CAB_ASSIGNED',
+            vehicleId:cabId,
+            endTime:null,
+            createdAt: {
+                [Op.gt]: new Date().setHours(0, 0, 0, 0),
+                [Op.lt]: new Date()
+            }
+        },
+        raw:true,
+    })
+
+    return !cabReserved
+}
+
+export const getNearestAvailableCabs = async ({ lat,long }) => {
+    const { QueryTypes } = db.vehicles.sequelize
+
+    //Check already reserved cabs
+    let bookedCabs = await db.bookings.findAll({
+        attributes:['vehicleId','id'],
+        where:{
+            status : 'CAB_ASSIGNED',
+            endTime:null,
+            createdAt: {
+                [Op.gt]: new Date().setHours(0, 0, 0, 0),
+                [Op.lt]: new Date()
+            }
+        }
+    })
+    let bookedCabIds = bookedCabs.map(cb=>cb.vehicleId)
+
+    let distanceQuery = `
+               (
+                               select *
+                               from distance(
+                                       ${lat}, ${long},
+                                       addr.lat, addr.long
+                                   )
+                           )       as disDiff
+            `
+    let sqlQuery = `
+                with distanceDiff as (
+                    select vh.*,
+                           vc.name as category,
+                           vs.name as sub_category,
+                           ${distanceQuery}
+                    from ${tables.address} addr
+                    inner join ${tables.vehicles} vh on vh.id = addr.item_id
+                    inner join ${tables.vehicleCategories} vc on vc.id = vh.vehicle_category_id
+                    inner join ${tables.vehicleSubCategories} vs on vs.id = vh.vehicle_sub_category_id
+                    where addr.type = 'VEHICLE'
+                    AND NOT (vh.id=ANY('{${bookedCabIds.join(',')}}'))
+                    order by disDiff ASC
+                    limit 20
+                    )
+                select * from distanceDiff;
+            `
+
+    return await db.vehicles.sequelize.query(
+        sqlQuery, {type: QueryTypes.SELECT}
+    )
+};
+
+//Return cab detail by their id
+export const getCustomerById = async (customerId, withPassword=false)=> {
+    let response = {}
+    try{
+        response.user  = convertDbResponseToRawResponse(
+            await db.customers.findOne({
+                where: {
+                    id: customerId
+                },
+                include: [
+                    {
+                        model: db.passport,
+                        as:'vehicle_category',
+                    }
+                ]
+            })
+        );
+
+        if(withPassword){
+            await db.passport.findOne({
```

Comment:
what does this line do?

Reply:
This is just one boolean flag for getting the password from another table, If passed true then we only return password data